Lib/ufo2ft/util.py

```diff
@@ -317,6 +320,13 @@ def classifyGlyphs(unicodeFunc, cmap, gsub=None):
             closeGlyphsOverGSUB(gsub, s)
             glyphs.update(s - neutralGlyphs)
 
+    if extra_substitutions:
+        for glyphs in glyphSets.values():
+            to_append = set()
+            for glyph in glyphs:
+                to_append |= extra_substitutions.get(glyph, set())
```

Comment:
Shouldn't this happen before we do a GSUB closure because these substituted glyphs could be substituted further?

Reply:
Hrm, probably.